### ISSUE TITLE
union type no type was matched and no error was reported.

### ISSUE THREAD

**Issue by WangZiChu199910252255** [microsoft/TypeScript#56635]
Use a combined type to define, and a single type will fail all tests, but no error will be reported when combined.

```ts
const example1: { A?: string } = { A: '', B: 12 }  // Object literals can only specify known properties, and 'B' is not in type '{ A?: string; }'.
const example2: { A?: string } & { B: number; C: string } = { A: '', B: 12 } // Type '{ A: string; B: number; }' cannot be assigned to type '{ A?: string; } & { B: number; C: string; }'. Property 'C' is missing from type '{ A: string; B: number; }' but is required in type '{ B: number; C: string; }'.
const example3: { A?: string } | ({ A?: string } & { B: number; C: string }) = { A: '' , C:'true'}
```

**Comment by NWYLZW** [comment]
The `example3` looks like the variable assign.
![image](https://github.com/microsoft/TypeScript/assets/51358815/64d09f67-ee36-4baa-a21e-df834ba5722b)

**Comment by MartinJohns** [comment]
Works as expected. Excess property checks do not work for union types, and the object literal is assignable to `{ A?: string }`.

**Comment by jcalz** [comment]
Duplicate of #20863, nice that the OP already tagged it as a such when opening the issue 🙃

**Comment by fatcerberus** [comment]
Yeah, but it only got tagged as a dupe because the OP used the 9998 issue template - which they apparently didn’t read because this is unrelated and you’re not supposed to post with that template anyway :wink:
